File: src/main/java/murilo/barbosa/array/exercises/TwoSum.rb
def two_sum(nums, target)
    map = Hash.new
    
    for i in 0..nums.size
        map[nums[i]] = i
    end

    for i in 0..nums.size
        difference = target - nums[i]
        target_index = map[difference]

        if target_index && target_index != i
            return [i, target_index]
        end
    end

    return []
end

File: src/main/java/murilo/barbosa/array/exercises/TwoSum.java
package murilo.barbosa.array.exercises;

import java.util.HashMap;

public class TwoSum {
    public int[] twoSum(int[] nums, int target) {

        var map = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length; i++) {
            map.put(nums[i], i);
        }

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];
            var targetIndex = map.get(complement);

            if (targetIndex != null && targetIndex != i) {
                return new int[]{i, map.get(complement)};
            }
        }

        return new int[]{};
    }
}
